tools/clang/test/CodeGenSPIRV/namespace.functions.hlsl

```diff
@@ -0,0 +1,99 @@
+// Run: %dxc -T ps_6_0 -E main
+
+// CHECK: OpName %AddRed "AddRed"
+// CHECK: OpName %A__AddRed "A::AddRed"
+// CHECK: OpName %A__B__AddRed "A::B::AddRed"
+// CHECK: OpName %A__B__AddBlue "A::B::AddBlue"
+// CHECK: OpName %A__AddGreen "A::AddGreen"
+// CHECK: OpName %A__createMyStruct "A::createMyStruct"
+// CHECK: OpName %A__myStruct_add "A::myStruct.add"
+
+// CHECK: [[v3f2:%\d+]] = OpConstantComposite %v3float %float_2 %float_2 %float_2
+// CHECK: [[v4f0:%\d+]] = OpConstantComposite %v4float %float_0 %float_0 %float_0 %float_0
+// CHECK: [[v3f0:%\d+]] = OpConstantComposite %v3float %float_0 %float_0 %float_0
+// CHECK: [[v3f1:%\d+]] = OpConstantComposite %v3float %float_1 %float_1 %float_1
+// CHECK: [[v3f3:%\d+]] = OpConstantComposite %v3float %float_3 %float_3 %float_3
+
+namespace A {
+
+  float3 AddRed() { return float3(0, 0, 0); }
+  float3 AddGreen();
+
+  namespace B {
+    typedef float3 f3;
+    float3 AddRed() { return float3(1, 1, 1); }
+    float3 AddBlue();
+  }  // end namespace B
+
+  struct myStruct {
+    int point1;
+    int point2;
+    int add() {
+      return point1 + point2;
+    }
+  };
+  
+  myStruct createMyStruct() {
+    myStruct s;
+    return s;
+  }
+}  // end namespace A
+
+
+float3 AddRed() { return float3(2, 2, 2); }
+
+float4 main(float4 PosCS : SV_Position) : SV_Target
+{
+// CHECK: {{%\d+}} = OpFunctionCall %v3float %AddRed
```

Comment:
Feel free to drop the `{{%\d+}}` part in the future since we are not interested in the return value here. No need to change for this one.